pkg/api/cluster_client.go

```diff
@@ -80,7 +81,11 @@ func (c ClusterClient) ReadByName(ctx context.Context, projectId, name string) (
 	}
 
 	if len(clusters.Data) != 1 {
-		return &models.Cluster{}, Error404
+		if most_recent {
+			sort.Slice(clusters.Data, func(i, j int) bool { return clusters.Data[i].CreatedAt.Seconds > clusters.Data[j].CreatedAt.Seconds })
+		} else {
+			return &models.Cluster{}, Error404
```

Comment:
Just suggest to return meaningful message here instead of `resource not found`. If it falls into this branch, it means that there is more than one resource with the same cluster name in the same project (which I admit it's rare).

Reply:
but it can happen, right? :) So, I'm updating the error message, thanks for the suggestion @kunliuedb 👍

Reply:
Wrote [here](https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/118#issuecomment-1385347383)